examples/ubo_tilemap/main.rs

```diff
@@ -0,0 +1,525 @@
+    }
+}
+
+// Actual tilemap data that makes up the elements of the UBO.
+// NOTE: It may be a bug, but it appears that
+// [f32;2] won't work as UBO data. Possibly an issue with
+// binding generation
+#[derive(Clone, Copy, Debug)]
+pub struct TileMapData {
+    data: [f32; 4]
+}
+
+impl TileMapData {
+    pub fn new_empty() -> TileMapData {
+        TileMapData { data: [0.0, 0.0, 0.0, 0.0] }
+    }
+    pub fn new(data: [f32; 4]) -> TileMapData {
+        TileMapData { data: data }
+    }
+}
+
+// Vertex data
+gfx_vertex!( VertexData {
+    a_Pos@ pos: [f32; 3],
+    a_BufPos@ buf_pos: [f32; 2],
+});
+
+// Shared uniforms used in the frag shader
+gfx_parameters!( Uniforms {
+    // projection stuff
+    u_Model@ model: [[f32; 4]; 4],
+    u_View@ view: [[f32; 4]; 4],
+    u_Proj@ proj: [[f32; 4]; 4],
+    // tilemap stuff
+    t_TileSheet@ tilesheet: gfx::shade::TextureParam<R>,
+    b_TileMap@ tilemap: gfx::handle::Buffer<R, TileMapData>,
+    u_WorldSize@ world_size: [f32; 3],
+    u_TilesheetSize@ tilesheet_size: [f32; 4],
+    u_TileOffsets@ offsets: [f32; 2],
+});
+
+// Abstracts the plane mesh and uniform data
+// Also holds a Vec<TileMapData> as a working data
+// set for consumers
+pub struct TileMapPlane<R> where R: Resources {
+    pub batch: Full<Uniforms<R>>,
+    pub data: Vec<TileMapData>,
+}
+
+impl<R> TileMapPlane<R> where R: Resources {
+    pub fn new<TFactory, TStream>(stream: &mut TStream, factory: &mut TFactory, width: usize, height: usize, tile_size: usize) -> TileMapPlane<R> where TFactory: Factory<R>, TStream: Stream<R> {
+        // charmap info
+        let half_width = (tile_size * width) / 2;
+        let half_height = (tile_size * height) / 2;
+        let total_size = width*height;
+
+        // tilesheet info
+        let tilesheet_bytes = &include_bytes!("scifitiles-sheet_0.png")[..];
+        let tilesheet_width = 14;
+        let tilesheet_height = 9;
+        let tilesheet_tilesize = 32;
+
+        let tilesheet_total_width = tilesheet_width * tilesheet_tilesize;
+        let tilesheet_total_height = tilesheet_height * tilesheet_tilesize;
+        // set up vertex data
+        let plane = Plane::subdivide(width, width);
+
+        // law out the vertices of the plane mesh based on the configured tile size information,
+        // setting the a_BufPos vertex data for the vertex shader (that ultimate gets passed through
+        // to the frag shader as a varying, used to determine the "current tile" and the frag's offset,
+        // which is used to calculate the displayed frag color)
+        let vertex_data: Vec<VertexData> = plane.shared_vertex_iter()
+            .map(|(raw_x, raw_y)| {
+                let vertex_x = half_width as f32 * raw_x;
+                let vertex_y = half_height as f32 * raw_y;
+
+                let u_pos = (1.0 + raw_x) / 2.0;
+                let v_pos = (1.0 + raw_y) / 2.0;
+                let tilemap_x = (u_pos * width as f32) as i32;
```

Comment:
You might want to use [floor()](https://doc.rust-lang.org/stable/std/primitive.f32.html#method.floor) here instead of converting it to an int and back.